src/HealthChecks.AzureServiceBus/AzureEventHubHealthCheck.cs

```diff
@@ -9,55 +10,72 @@ public class AzureEventHubHealthCheck : IHealthCheck
+        private static readonly ConcurrentDictionary<string, EventHubProducerClient> _clientConnections = new();
+
+        private readonly string? _connectionString;
+        private readonly string? _endpoint;
+        private readonly string? _eventHubName;
+        private readonly TokenCredential? _tokenCredential;
+
+        private string? _connectionKey;
+
+        private string ConnectionKey =>
+            _connectionKey ??= _connectionString ?? $"{_endpoint}_{_eventHubName}";
 
         public AzureEventHubHealthCheck(string connectionString, string eventHubName)
         {
-            Guard.ThrowIfNull(connectionString, true);
+            Guard.ThrowIfNull(connectionString, true);
             Guard.ThrowIfNull(eventHubName, true);
 
-            _eventHubConnectionString = connectionString.Contains(ENTITY_PATH_SEGMENT) ? connectionString : $"{connectionString};{ENTITY_PATH_SEGMENT}{eventHubName}";
+            _connectionString = connectionString.Contains(ENTITY_PATH_SEGMENT) ? connectionString : $"{connectionString};{ENTITY_PATH_SEGMENT}{eventHubName}";
 
-            if (!_eventHubConnections.ContainsKey(_eventHubConnectionString))
-            {
-                _eventHubConnections.TryAdd(_eventHubConnectionString, new EventHubProducerClient(_eventHubConnectionString));
+            if (!_clientConnections.ContainsKey(_connectionString))
+            {
+                _clientConnections.TryAdd(_connectionString, new EventHubProducerClient(_connectionString));
             }
-        }
+        }
 
         public AzureEventHubHealthCheck(EventHubConnection connection)
         {
-            Guard.ThrowIfNull(connection);
+            Guard.ThrowIfNull(connection);
+
+            _connectionString = $"{connection.FullyQualifiedNamespace};{ENTITY_PATH_SEGMENT}{connection.EventHubName}";
+
+            if (!_clientConnections.ContainsKey(_connectionString))
+            {
+                _clientConnections.TryAdd(_connectionString, new EventHubProducerClient(connection));
+            }
+        }
+
+        public AzureEventHubHealthCheck(string endpoint, string eventHubName, TokenCredential tokenCredential)
+        {
+            _endpoint = Guard.ThrowIfNull(endpoint, true);
+            _eventHubName = Guard.ThrowIfNull(eventHubName, true);
+            _tokenCredential = Guard.ThrowIfNull(tokenCredential);
 
-            _eventHubConnectionString = $"{connection.FullyQualifiedNamespace};{ENTITY_PATH_SEGMENT}{connection.EventHubName}";
-
-            if (!_eventHubConnections.ContainsKey(_eventHubConnectionString))
-            {
-                _eventHubConnections.TryAdd(_eventHubConnectionString, new EventHubProducerClient(connection));
+            string key = ConnectionKey;
+            if (!_clientConnections.ContainsKey(key))
+            {
+                _clientConnections.TryAdd(key, new EventHubProducerClient(_endpoint, _eventHubName, _tokenCredential));
```

Comment:
`TryAdd` _may_ return `false`. I suggest you to look into `internal static class ClientCache` under `HealthChecks.CosmosDb` namespace.

Reply:
Added ClientCache 👍🏻